handlers/handlers.go

```diff
@@ -17,50 +17,38 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
-type StreamInfo struct {
-	callbackUrl string
-}
-
 type CatalystAPIHandlersCollection struct {
 	MistClient  MistAPIClient
-	StreamCache map[string]StreamInfo
+	StreamCache *StreamCache
```

Comment:
I think we need to add somewhere the cleanup for this `StreamCache`. Currently we just add values here, but if there is any error in this whole process, then we may end up with some streams handing in Mist. 

Saying that, I think it's not part of this PR, but I suggest to create a separate GH Issue for it.